plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/idle/ServerIdleDetector.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine.idle;
+
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.che.api.core.event.ServerIdleEvent;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.core.notification.EventSubscriber;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.shared.dto.event.WorkspaceStatusEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+
```

Comment:
The similar functionality we had in Codenvy/partial Che. And I think we can potentially port/enable in Eclipse Che. See more 
https://github.com/codenvy/codenvy/tree/master/wsagent/codenvy-hosted-workspace-activity-agent
https://github.com/codenvy/codenvy/tree/master/wsmaster/codenvy-hosted-workspace-activity
https://github.com/eclipse/che/blob/master/agents/go-agents/core/activity/activity.go#L46
https://github.com/eclipse/che/blob/master/agents/go-agents/terminal-agent/main.go#L136
https://github.com/codenvy/codenvy/blob/master/plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/agent/CodenvyAgentModule.java#L43
I prefer that we think to port it in che instead of creating parallel functionality. The profits that I see.
 - General API for many agents like go terminal, ws-agent
 - Not connected to some backend-specific